src/libraries/ObservationLib.sol

```diff
@@ -77,13 +80,7 @@ library ObservationLib {
 
       beforeOrAtIndex = uint16(RingBufferLib.wrap(currentIndex, _cardinality));
       beforeOrAt = _observations[beforeOrAtIndex];
-      uint48 beforeOrAtTimestamp = beforeOrAt.timestamp;
-
-      // We've landed on an uninitialized timestamp, keep searching higher (more recently).
-      if (beforeOrAtTimestamp == 0) {
```

Comment:
Why don't we need this condition anymore?

Reply:
Because we don't have gaps in our observations like Uniswap does; all indices lower than the cardinality will be populated.